addons/website/static/src/js/menu/translate.js

```diff
@@ -246,6 +276,19 @@ var TranslatePageMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
                 self._markTranslatableNodes();
                 this.$translations.filter('input[type=hidden].o_translatable_input_hidden').prop('type', 'text');
             },
+            processRecordsCallback(record, el) {
+                const tocMainEl = el.closest('.s_table_of_content_main');
+                const headerEl = el.closest('h1, h2');
+                if (!tocMainEl || !headerEl) {
+                    return;
+                }
+                const headerIndex = [...tocMainEl.querySelectorAll('h1, h2')].indexOf(headerEl);
+                const tocMenuEl = tocMainEl.closest('.s_table_of_content').querySelectorAll('.table_of_content_link > span')[headerIndex];
+                if (tocMenuEl.textContent !== headerEl.textContent) {
+                    tocMenuEl.textContent = headerEl.textContent;
```

Comment:
We already have a way to sync titles in regular edit mode, can't we somehow reuse/share it in translate mode? (Didn't check so open question)

Reply:
The edit mode sync mechanism is a full regeneration of the table of content on any change seen by a `MutationObserver` inside the "content" part of the snippet - it even generates new ids. Here we want to show that only a very specific text has changed, so I do not think it can be reused.
Replacing the other mechanism by this one also does not sound great because this can only be used to update existing labels - it would not work for adding missing ones or removing some.